### ISSUE TITLE
Create test script for models

### ISSUE THREAD

**Issue by skekre98** [skekre98/NBA-Search#28]
**Description of feature/enhancement**
Create a script to test the accuracy of the models such as with `query_classifier`.

**Description of implementation**
Perhaps started with a few from query.csv, but also add in a few obscure queries that may be harder for the model to predict.

**Why this is a good feature/enhancement**
Need a way to confirm model accuracy on something other than training data.

**Additional context**
This framework should ideally also be able to be used on any other models required in the future.

**Comment by JohnnyUrosevic** [comment]
Hello I want to work on this, how would you recommend I go about this. I assume you want me to make a python script that takes in test cases and some sort of expected value? I assume I would have to go about making these expected values by hand? Let me know if there's anything more specific you had in mind.

**Comment by ashkankamyab** [comment]
@skekre98 
I am also interested to work on this.

**Comment by skekre98** [comment]
@JohnnyUrosevic Yes, this is along the lines of what I was thinking. The script should ideally test the models on queries different than those present in the training set. Perhaps create a directory name _tests_ under `../inference/models` and add the script there. I will go ahead an assign the issue over to you.

**Comment by skekre98** [comment]
@ashkankamyab Unfortunately @JohnnyUrosevic was first to comment. I don't think this issue will require more than one person's attention. However, would you be interested in undertaking #33? The issue is very similar as we need to test the results of a machine learning algorithm that was implemented earlier. If so, please comment on the issue and I will assign it to you ☺️.

**Comment by ashkankamyab** [comment]
> @ashkankamyab Unfortunately @JohnnyUrosevic was first to comment. I don't think this issue will require more than one person's attention. However, would you be interested in undertaking #33? The issue is very similar as we need to test the results of a machine learning algorithm that was implemented earlier. If so, please comment on the issue and I will assign it to you ☺️.

Thanks @skekre98 
I will take a closer look into #33 :-)

**Comment by skekre98** [comment]
@ashkankamyab would you mind commenting on #33? Github won't let me assign it to you until you are part of the thread 😅.

**Comment by ashkankamyab** [comment]
I will comment as soon as I am ready to work on it :-)

**Comment by skekre98** [comment]
@JohnnyUrosevic any update on this issue?

**Comment by JohnnyUrosevic** [comment]
@skekre98 Hello, I've been unexpectedly extremely busy this month and haven't had a chance to work on it. If you want to assign it to someone else that is okay, otherwise I will look for a time to work on it

**Comment by kelaaron** [comment]
Hi @skekre98, I can take this issue. It will probably take me a couple weeks to get familiar with everything but I'd love to work on it!

**Comment by skekre98** [comment]
@kelaaron Sure, that would be great!

**Comment by kelaaron** [comment]
@skekre98 I've started thinking about implementation for this. If I understand correctly, this test script should take in model to test (such as query_classifier.py) and test cases (hypothetically made of values from query.csv, as well as other obscure queries for query_classifier), run those tests on the given model, and return the percentage of queries which had their respective class accurately predicted.

Does this implementation make sense? I'd love to hear anything I might be missing or could improve on. I plan on beginning to work on this sometime in the next few days. 😄

**Comment by skekre98** [comment]
Hi @kelaaron, really sorry about the late response. I must've missed your message. This sounds like a good starting point! One thing to keep in the back of your mind is that this framework should be expandable to analyze other model attributes beyond just accuracy in the future such as latency or validation results. This implementation definitely sounds like it's on the right track.
